File: server/internal/models/route.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
	"time"
)

type Route struct {
	ID          primitive.ObjectID `json:"id" bson:"_id"`
	Name        string             `json:"name" bson:"name" validate:"required,min=2,max=100"`
	Points      [][]float32        `json:"points" bson:"points" validate:"required"`
	Coordinates [][]float32        `json:"coordinates" bson:"coordinates" validate:"required"`
	CreatedAt   time.Time          `json:"created_at,omitempty" bson:"created_at"`
	UserID      primitive.ObjectID `json:"user_id" bson:"user_id"`
}

File: client/src/models/route.ts
export interface Route {
  id: string | any,
  name: string,
  points: Array<Array<string>>,
  coordinates: Array<Array<number>>,
  created_at: Date,
  user_id: string
}

export function GetRouteStartPoint(route: Route): Array<number> {
  return [Number(route.points[0][1]), Number(route.points[0][0])] 
}